subseas_flex_fcst/maproom.py

```diff
@@ -0,0 +1,516 @@
+    return center_of_the_map, lat_min, lat_max, lon_min, lon_max, lat_label, lon_label
+
+
+def get_coords(click_lat_lng):
+    if click_lat_lng is not None:
+        return click_lat_lng
+    else:
+        fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
+        return [(fcst_mu.Y[0].values+fcst_mu.Y[-1].values)/2, (fcst_mu.X[0].values+fcst_mu.X[-1].values)/2]
+
+
+def round_latLng(coord):
+    value = float(coord)
+    value = round(value, 1)
+    return value
+
+
+@APP.callback(Output("map", "click_lat_lng"), Input("submitLatLng","n_clicks"), State("latInput", "value"), State("lngInput", "value"))
+def inputCoords(n_clicks,latitude,longitude):
+    if latitude is None:
+        return None
+    else:
+        lat_lng = [latitude, longitude]
+        return lat_lng
+
+
+@APP.callback(Output("layers_group", "children"), Input("map", "click_lat_lng"))
+def map_click(click_lat_lng):
+    lat_lng = get_coords(click_lat_lng)
+
+    return dlf.Marker(
+        position=lat_lng, children=dlf.Tooltip("({:.3f}, {:.3f})".format(*lat_lng))
+    )
+
+
+@APP.callback(
+    Output("percentile_style", "style"),
+    Output("threshold_style", "style"),
+    Input("variable", "value")
+)
+def display_relevant_control(variable):
+
+    displayed_style={
+        "position": "relative",
+        "width": "190px",
+        "display": "flex",
+        "padding": "10px",
+        "vertical-align": "top",
+    }
+    if variable == "Percentile":
+        style_percentile=displayed_style
+        style_threshold={"display": "none"}
+    else:
+        style_percentile={"display": "none"}
+        style_threshold=displayed_style
+    return style_percentile, style_threshold
+
+
+@APP.callback(
+    Output("phys_units", "children"),
+    Input("submitLatLng","n_clicks"),
+)
+def read_units(toto):
+
+    fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
+    fcst_mu_name = list(fcst_mu.data_vars)[0]
+    return [" "+fcst_mu[fcst_mu_name].attrs["units"]]
+
+
+@APP.callback(
+    Output("cdf_graph", "figure"),
+    Output("pdf_graph", "figure"),
+    Input("map", "click_lat_lng"),
+)
+def local_plots(click_lat_lng):
+
+    lat, lng = get_coords(click_lat_lng)
+
+    # Reading
+    fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
+    fcst_mu_name = list(fcst_mu.data_vars)[0]
+    fcst_mu = fcst_mu[fcst_mu_name]
+    fcst_var = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_var_file"])))
+    fcst_var_name = list(fcst_var.data_vars)[0]
+    fcst_var = fcst_var[fcst_var_name]
+    obs = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["obs_file"])))
+    obs_name = list(obs.data_vars)[0]
+    obs = obs[obs_name]
+    if CONFIG["y_transform"]:
+        hcst = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["hcst_file"])))
+        hcst_name = list(hcst.data_vars)[0]
+        hcst = hcst[hcst_name]
```

Comment:
In theory it can; though I'm honestly not sure of an easy mechanism for generating a multi-field CPTv10 file. Angel and Simon want to put the ensemble averaging into CPT, so itll probably be needed at that point. 

I'm happy to put this 3-liner into an open_cptdataarray in a future release, with the convention that you can only open something with one field